contracts/token/ERC721/extensions/ERC721Burnable.sol

```diff
@@ -19,9 +19,8 @@ abstract contract ERC721Burnable is Context, ERC721 {
      * - The caller must own `tokenId` or be an approved operator.
      */
     function burn(uint256 tokenId) public virtual {
-        if (!_isApprovedOrOwner(_msgSender(), tokenId)) {
-            revert ERC721InsufficientApproval(_msgSender(), tokenId);
-        }
-        _burn(tokenId);
+        // Setting an "auth" arguments enables the `_isApproved` check which verifies that the token exists
+        // (from != 0). Therefore, it is not needed to verify that the return value is not 0 here.
+        _update(address(0), tokenId, _msgSender());
```

Comment:
Even though `_update` is a lower level function it is still available and can be used, so I don't think we need the more complex `_burn` definition. If we added that function, it sounds like we should also add `_mint(..., auth)` and `_transfer(..., auth)`, etc., and I don't think we should.

Reply:
Then I'd advocate for removing the `_burn` function. The fact we're not using the `_burn` function here because it doesn't meet our needs is imo strong evidence that the function would also be not enough for several use cases.

We don't need `auth` neither in `_transfer` or `_mint` because the former don't require operator check because it checks the `from`, and the later doesn't require operator check because it doesn't belong to anyone in the first place. I think `_burn` needs an `auth` because it's destroying someone else's property and there's no `from` check.

Reply:
The current `_mint`, `_burn` and `_transfer` are designed to copy the 4.X behavior. This is so the migration would be simplified. Indeed, it is not as powerfull as the new `_update`.

Its a balance betten simplifying migration through functions that are not suited for all usecases, or removing them and force users to addapt to the new pattern.